File: sync/src/main/scala/com/fan/mysql/event/impl/FormatDescriptionEvent.java
package com.fan.mysql.event.impl;


import com.fan.mysql.event.AbstractEvent;
import com.fan.mysql.event.EventHeader;

public class FormatDescriptionEvent extends AbstractEvent {

    private int binlogVersion;
    private String serverVersion;
    private long createTime;
    private int commonHeaderLen;
    private short[] postHeaderLen;

    public FormatDescriptionEvent(EventHeader header) {
        super(header);
    }

    public FormatDescriptionEvent(int binlogChecksum) {
        super(new EventHeaderImpl());
        header.setChecksumAlg(binlogChecksum);
    }

    public int getBinlogVersion() {
        return binlogVersion;
    }

    public void setBinlogVersion(int binlogVersion) {
        this.binlogVersion = binlogVersion;
    }

    public String getServerVersion() {
        return serverVersion;
    }

    public void setServerVersion(String serverVersion) {
        this.serverVersion = serverVersion;
    }

    public long getCreateTime() {
        return createTime;
    }

    public void setCreateTime(long createTime) {
        this.createTime = createTime;
    }

    public int getCommonHeaderLen() {
        return commonHeaderLen;
    }

    public void setCommonHeaderLen(int commonHeaderLen) {
        this.commonHeaderLen = commonHeaderLen;
    }

    public short[] getPostHeaderLen() {
        return postHeaderLen;
    }

    public void setPostHeaderLen(short[] postHeaderLen) {
        this.postHeaderLen = postHeaderLen;
    }

}

File: async/src/main/scala/com/fan/mysql/async/binlog/event/impl/FormatDescriptionEvent.scala
package com.fan.mysql.async.binlog.event.impl

import com.fan.mysql.async.binlog.event.{AbstractEvent, EventHeader}

class FormatDescriptionEvent(header: EventHeader) extends AbstractEvent(header) {
  private[this] var binlogVersion: Int = _
  private[this] var serverVersion: String = _
  private[this] var createTime: Long = _
  private[this] var commonHeaderLen: Int = _
  private[this] var postHeaderLen: Array[Short] = _

  def this(binlogChecksum: Int) = {
    this(EventHeader(checksumAlg = binlogChecksum))
  }

  def getBinlogVersion: Int = binlogVersion

  def setBinlogVersion(binlogVersion: Int): Unit = {
    this.binlogVersion = binlogVersion
  }

  def getServerVersion: String = serverVersion

  def setServerVersion(serverVersion: String): Unit = {
    this.serverVersion = serverVersion
  }

  def getCreateTime: Long = createTime

  def setCreateTime(createTime: Long): Unit = {
    this.createTime = createTime
  }

  def getCommonHeaderLen: Int = commonHeaderLen

  def setCommonHeaderLen(commonHeaderLen: Int): Unit = {
    this.commonHeaderLen = commonHeaderLen
  }

  def getPostHeaderLen: Array[Short] = postHeaderLen

  def setPostHeaderLen(postHeaderLen: Array[Short]): Unit = {
    this.postHeaderLen = postHeaderLen
  }

}
